### PR TITLE
Export model

### PR DESCRIPTION
**Pull request by tomer-algo**
# Intro
  - Export a PyTorch Skeleton Model for serving using TorchServe
# Project
  - Suggested-Skeleton
  - Smart Sheets
# Implemented Sections
  - First implementation for serving model in TorchServe
  - Converting to TorchScript model (production oriented)
  - Updating demo inference scripts
  - Did it require refactoring in other locations as well? No
# Tests Description
  - Export Model & run inference on TorchServe locally
  - Export Model & run batch inference on TorchServe locally
# Undeveloped Sections
  - 
# What's New
  - Batch Inference including pre/post on PyTorch Model
# Checklist
  - [x] Module tested fully after last commit
  - [x] Scheduled a PR meeting (if needed)
  - [ ] Documentation (comments or a dedicated document)
### Tested platforms
  - [x] Local
  - [ ] Cloud (Sagemaker)
  - [ ] Staging
  - [ ] Not relevant
### Tested Cameras
  - [ ] Gen 1.5
  - [ ] Gen 2
  - [x] Not relevant
# Comments
### Included Features
  - First implementation for serving model in TorchServe
    - Exporting script
    - Skeleton handler
      - Handling pre/post processing for skeleton model
    - Converting to TorchScript model (production oriented)
    - Updating demo inference scripts
### Infrastructure Dependencies
  - Pass initial model to AI-Infra (changes might be required from AI-Algo side)
### Tech Comments
  - 
### Testing Instructions
  - How to skeleton inference in TorchServe: https://docs.google.com/document/d/18TOAEodIMwRWKnJIjXBFvGpwRB9Px_mYmV02ua245L4/edit?usp=sharing
### Knowledge
| / | Link |
| ------ | ------ |
| Monday | https://nanit.monday.com/boards/912051282/pulses/1428805988 |
| TLD |  |
| Tech Spec | https://docs.google.com/document/d/1I-nFbP9ogt4qjc6rG6KAgeErb1NiQjf4Lk8LWF0O2-M/edit?usp=sharing |
### Todos
  - Check Option to improve pre/post for batch inference (Tensor instead of Matrix)

### CODE REVIEW

File: export/skeleton_handler.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        return model
+
+    def _load_torchscript_model(self, model_pt_path):
+        """Loads the PyTorch model and returns the NN model object.
+        Args:
+            model_pt_path (str): denotes the path of the model file.
+        Returns:
+            (NN Model Object) : Loads the model object.
+        """
+        return torch.jit.load(model_pt_path, map_location=self.device)
```
Comment (tomer-algo) [inline export/skeleton_handler.py:218]: check if method exist in parent class

File: lib/core/inference.py [hunk 2]
```diff
@@ -77,3 +81,45 @@ def get_final_preds(config, batch_heatmaps, center, scale):
         )
 
     return preds, maxvals
+
+def get_pose_estimation_prediction(pose_model, device, cfg, images, centers, scales, transform=None):
+    rotation = 0
```
Comment (tomer-algo) [inline lib/core/inference.py:86]: add comment for batching 
assumption one element in image

**Comment by tomer-algo** [comment]
Add to documentation blogs on torchserve

File: export/skeleton_handler.py [hunk 3]
```diff
@@ -81,54 +81,68 @@ def preprocess(self, data):
...
+                image = base64.b64decode(image)
+
+            # If the image is sent as bytesarray
+            if isinstance(image, (bytearray, bytes)):
+                image = Image.open(io.BytesIO(image))
+                image = np.array(image.convert('RGB'))
+            else:
+                raise ValueError('image is not sent as bytesarray')
+
+            baby_bbox_list = eval(d['baby'])
```
Comment (Nimrods) [inline export/skeleton_handler.py:107]: Why do you need `eval()`? why not just `baby_bbox_list = d['baby']`?

Same goes for head

Comment (tomer-algo) [reply export/skeleton_handler.py:107]: the input is a string (from curl command) changing it into a list

**Review by Nimrods — APPROVED** [review]

**Review by felixNan — APPROVED** [review]

**Pull request merged**